src/allocator/allocator.c

```
@@ -280,31 +303,34 @@ int add_chunk_async(CUdeviceptr *address, size_t size, CUstream hStream) {
         return res;
     }
     *address = e->entry->address;
+    /* cuMemAllocAsync draws from the device's default pool. */
     CUmemoryPool pool;
     res = CUDA_OVERRIDE_CALL(cuda_library_entry,cuDeviceGetMemPool,&pool,dev);
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)